pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
 
 
+def decompose(x, type, f, filter=None):
+    """
+
+    Parameters
+    ----------
+    x : np.array
+        The time series of which the trend, seasonal, and noise/random
+        components will be extracted.
+
+    type: str
+        The type of decomposition that will be performed - 'multiplicative' or
+        'additive'. We would use 'multiplicative' generally when we see an
+        increasing trend. We use 'additive' when the trend is relatively
+        stable over time.
+
+    f: int
+        The frequency in terms of number of observations. This behaves
+        similarly to R's frequency for a time series (ts).
+
+    filter: np.array
+        A filter by which the convolution will be performed.
+
+    Returns
+    -------
+    decomposed_tuple : namedtuple
+        A named tuple with `x`, `trend`, `seasonal`, and `random` components
+        where `x` is the input signal, `trend` is the overall trend, `seasonal`
+        is the seasonal component, and `random` is the noisy component. The
+        input signal `x` can be mostly reconstructed by the other three
+        components with a number of points missing equal to `f`.
+
+    Notes
+    -----
+    This function is generally used in conjunction with
+    :func:`pmdarima.utils.visualization.decomposed_plot`,
+    which plots the decomposed components. Also there are two example notebooks
+    within the `examples` folder which mirror existing R examples.
+
+     References
+    ----------
+    .. [1] Example of decompose using both multiplicative and additive types:
+           https://anomaly.io/seasonal-trend-decomposition-in-r/index.html
+
+    .. [2] R documentation for decompose:
+           https://www.rdocumentation.org/packages/stats/versions/3.6.1/topics/decompose
+
+
+    """
+
+    multiplicative = "multiplicative"
+    additive = "additive"
+
+    # Helper function to stay consistent and concise based on 'type'
+    def _decomposer_helper(a, b, type):
+        if type == multiplicative:
+            return a / b
+        else:
+            return a - b
+
+    # Since R's ts class has a frequency as input I think this it acceptable
+    # to ask the user for the frequency.
+    try:
+        assert int(f) > 1 and isinstance(f, int)
+    except (ValueError, AssertionError):
+        raise ValueError("'f' should be a positive integer")
+
+    if filter is None:
+        filter = np.ones((f,)) / f
+
+    # We only accept the values in multiplicative or additive
+    if type not in (multiplicative, additive):
+        err_msg = "'type' can only take values '{}' or '{}'"
+        raise ValueError(err_msg.format(multiplicative, additive))
+
+    # There needs to be at least 2 periods. This is due to the behavior of
+    # convolutions and how they behave with respect to losing endpoints
+    if (x.shape[0] / f) < 2:
+        raise ValueError("time series has no or less than 2 periods")
+
+    # Take half of f for the convolution / sma process.
+    half_f = int(f / 2.)
```

Comment:
Unimportant nitpick, since you know `m` (`f`) is an int now, you could just do `f // 2`

Reply:
Done!